### PR TITLE
Remove the build time jar dependency installation and embedded the jars instead

### PR DESCRIPTION
**Pull request by purbon**
Clean up jar dependencies code and so we end embedded dependencies  the same gem. To do so I introduced a common code to load jars so we aim to simplify the management of jars in the long run. This loading code can be extracted to a common gem when agreed on a way to do this task.

Version 0.1.5 bump

Remove the vendor/ dir from the files added in the gemspec as this directory is no longer in use.

@ph comments? ideas? naming for the common loading module?

### CODE REVIEW

File: lib/logstash-input-log4j_jars.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+module LogStash
+  module JarDependencies
+
```
Comment (ph) [inline lib/logstash-input-log4j_jars.rb]: useless space

**Comment by ph** [comment]
minor comment but LGTM! I love simplicity.

**Comment by wiibaa** [comment]
@purbon @ph is it not more or less what [logstash::environment](https://github.com/elastic/logstash/blob/master/lib/logstash/environment.rb#L58) is doing for loading elasticsearch jar and could be generalized as proposed in https://github.com/elastic/logstash/pull/1301

**Comment by purbon** [comment]
Yes @wiibaa, this is the same thing and the basic idea is to generalize this to follow the same pattern everywhere where a jar is required.

**Comment by ph** [comment]
@wiibaa exactly, we are going to a more simpler flow and one less dependencies system to manage..

**Comment by wiibaa** [comment]
@purbon @ph thanks for the explanations and sorry for the silly question but shouldn't there be a `require "lib/logstash-input-log4j_jars"` somewhere ?

**Comment by purbon** [comment]
@wiibaa https://github.com/logstash-plugins/logstash-input-log4j/blob/master/lib/logstash/inputs/log4j.rb#L9 is there :-)

**Comment by ph** [comment]
@wiibaa :beers:  :)

**Comment by wiibaa** [comment]
ok got it, the require was already there, but not the file because it was generated by jar-dependendies at build time...
@ph I should be the one offering the beers ;)
@purbon thanks for the lesson

**Comment by ph** [comment]
@wiibaa If I ever go to europe I'll accept them.

**Comment by colinsurprenant** [comment]
oh, I'm in Europe now, I can take care of that :innocent:

**Comment by jsvd** [comment]
LGTM (didn't test in a logstash installation)

**Pull request closed without merging**